NearestExonJB.pm

```diff
@@ -106,9 +115,39 @@ sub run {
   my $loc_string = sprintf("%s:%s-%i-%i", $trv->transcript_stable_id, $vf->{chr} || $vf->seq_region_name, $vf->{start}, $vf->{end});
 
   if(!exists($self->{_cache}) || !exists($self->{_cache}->{$loc_string})) {
-    my $exons = $trv->_overlapped_exons;
+    my $exons = $trv->_overlapped_exons; # intronic variants do not overlap any exon
     my %dists;
     my $min = $CONFIG{max_range};
+
+    # For option --intronic, fetch the list of exons with different method
+    # Do not take into account the max_range
+    if(scalar @{$exons} == 0 && $CONFIG{intronic} == 1) {
+      my $intron_numbers = $trv->intron_number();
+      my $consequences = join(",", map { $_->SO_term } @{$tva->get_all_OverlapConsequences});
+
+      if(defined $intron_numbers && $consequences =~ /intron/) {
+        $exons = $trv->_sorted_exons;
+        my ($intron_number, $total_number) = split(/\//, $intron_numbers);
+ 
+        # Get the number of exons before and after the intron
+        my $exon_before = $intron_number;
+        my $exon_after = $intron_number + 1;
+
+        my @exons_tmp;
+        # Reverse strand we get the exons from the end of the list
+        if($tva->transcript->strand < 0) {
```

Comment:
Suggesting that we just directly modify the $exons like this
```
if ($tva->transcript->strand < 0) {
@$exons = ($exons->[-$intron_number], $exons->[-($intron_number + 1)]);
}
else {
@$exons = ($exons->[$intron_number - 1], $exons->[$intron_number]);  
}
```

Reply:
Changing the exons list like this affects other variants in the same transcript. It's weird.
You can test with this input:
```
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO
11      1587765 .       CTTTTTTTTTTTTTTTTTTTT   CTTTTTTTTTTTTTTTTTTTTTTTTTTTT   .      .       .
11      1613675 .       C       T       . .       .
```

Reply:
For the first variant, `$exons = $trv->_sorted_exons` returns 3 exons.
For the second variant `$exons = $trv->_sorted_exons` returns only 2 exons which are the exons selected in the previous variant.
`$trv` = ENST00000792906 for both variants.

Reply:
I agree.